util/ranger/detacher.go

```diff
@@ -200,42 +199,24 @@ func removeAccessConditions(conditions, accessConds []expression.Expression) []e
 	return filterConds
 }
 
-// DetachCondsForSelectivity detaches the conditions used for range calculation from other useless conditions.
+// DetachCondsForSelectivity detaches the access conditions used for range calculation.
```

Comment:
ExtractAccessCondition is a better name.

Reply:
Hi @tiancaiamao  

when reading the source code, I am a little bit confused about the purpose of `ExtractAccessCondition`. Is this method used to filter all expressions relating to specific column?

1. Suppose here we have a SQL statement

```
select * from users where name = "Ryan" and age > 30 and age < 35 and age in (31, 39, 40, 24)
```

2. When calling this method `ExtractAccessCondition(expressions, column_a's id)`, the following expression should be filtered.

- age > 30
- age < 35
- age in (31, 39, 40, 24)


Is my assumption correct?

cc @winoros

Reply:
It should be `ExtractAccessConditionsForColumn` now.
e.g. `where a > 30 and a < 100 and ((a > 50 and c < 10) or (a < 30 and b > 20))`
It would extract `a > 30 and a < 100`.

Reply:
Thanks for your explanation. It's very helpful.